components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java

```diff
@@ -2560,4 +2561,25 @@ public static boolean isContinueOnClaimHandlingErrorAllowed() {
         // If config is empty or not a boolean value, the property must be set to the default value which is true.
         return !Boolean.FALSE.toString().equalsIgnoreCase(continueOnClaimHandlingErrorValue);
     }
+
+    public static boolean isAbsoluteURI(String uri) {
+
+        if (StringUtils.isBlank(uri)) {
+            if (log.isDebugEnabled()) {
+                log.debug("URI is empty.");
+            }
+            return true;
+        }
+
+        try {
+            final URI uriObj = new URI(uri);
+            return uriObj.isAbsolute();
+        } catch (URISyntaxException e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Unable to process the URI: " + uri, e);
+            }
+        }
+        // Default behavior of a URI is expected to be absolute;
+        return true;
```

Comment:
If the uri throws an `URISyntaxException` it will return `true`. Is it expected behavior?

Reply:
fixed